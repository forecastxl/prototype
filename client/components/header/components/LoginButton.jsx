import React, { PropTypes } from 'react'
import { Link } from 'react-router-dom'
import muiThemeable from 'material-ui/styles/muiThemeable'
import FlatButton from 'material-ui/FlatButton'

export function LoginButton({ muiTheme }) {
  return (
    <Link to="/login">
      <FlatButton style={{ color: muiTheme.palette.alternateTextColor }} label="Login" />
    </Link>
  )
}

LoginButton.propTypes = {
  muiTheme: PropTypes.object.isRequired
}

export default muiThemeable()(LoginButton)
