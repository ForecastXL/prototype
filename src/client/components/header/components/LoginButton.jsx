import React from 'react'
import { object } from 'prop-types'
import { Link } from 'react-router-dom'
import muiThemeable from 'material-ui/styles/muiThemeable'
import FlatButton from 'material-ui/FlatButton'

export function DumbLoginButton({ muiTheme }) {
  return (
    <Link to="/login">
      <FlatButton style={{ color: muiTheme.palette.alternateTextColor }} label="Login" />
    </Link>
  )
}

DumbLoginButton.propTypes = {
  muiTheme: object.isRequired
}

export default muiThemeable()(DumbLoginButton)
