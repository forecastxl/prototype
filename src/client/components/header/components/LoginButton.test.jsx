import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { DumbLoginButton } from './LoginButton'

describe('<DumbLoginButton />', () => {
  it('renders correctly', () => {
    const theme = {
      palette: {
        alternateTextColor: 'blue'
      }
    }
    const wrapper = shallow(<DumbLoginButton muiTheme={theme} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
