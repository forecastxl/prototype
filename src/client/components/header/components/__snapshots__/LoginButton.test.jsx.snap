// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DumbLoginButton /> renders correctly 1`] = `
<Link
  replace={false}
  to="/login"
>
  <FlatButton
    disabled={false}
    fullWidth={false}
    label="Login"
    labelPosition="after"
    labelStyle={Object {}}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
    style={
      Object {
        "color": "blue",
      }
    }
  />
</Link>
`;
